technologies/convince/README.md

```diff
@@ -3,14 +3,248 @@
+
+Lets's look at the property we get as an output in some more detail:
+
+```txt
+Pmin=? [F ((topic_level_msg.ros_fields__data <= 0) & topic_level_msg.valid)]
+```
+
+Here
+
+- `Pmin=?` means that we are looking for the minimum probability that the property holds.
+- `F` is the temporal operator "finally" (the property holds at some point in the future ([LTL](https://en.wikipedia.org/wiki/Linear_temporal_logic))
+- `((topic_level_msg.ros_fields__data <= 0) & topic_level_msg.valid)` is the logical expression that we are looking for:
+  - `topic_level_msg.ros_fields__data <= 0` means that the battery level is less than or equal to 0. Note that this refers to the ROS topic `level` and it's field `data`.
+  - `topic_level_msg.valid` is something we add to the model to make sure the data has been sent at least once, otherwise we may check against uninitialized data.
+
+### Task 1.2: Checking other properties
+
+Without changing the model, you can also check the other properties by specifying them in the `smc_storm` command:
+
+```bash
+smc_storm --model main.jani --properties-names battery_over_depleted
+```
+
+This will check whether the battery level will eventually reach -1 or lower.
+The output should look like this:
+
+```txt
+CONVINCE Statistical Model Checker
+Checking model: main.jani
+Property "battery_over_depleted": Pmin=? [F ((topic_level_msg.ros_fields__data <= -1) & topic_level_msg.valid)];
+Result: 1
+```
+
+This is not what we expected, as the battery level should never reach -1.
+Something is wrong with the model of the battery drainer.
+
+You can also check the third property:
+
+```bash
+smc_storm --model main.jani --properties-names alarm_on
+```
+
+This should output:
+
+```txt
+CONVINCE Statistical Model Checker
+Checking model: main.jani
+Property "alarm_on": Pmin=? [(!(topic_level_msg.valid) | (topic_level_msg.ros_fields__data >= 29)) U (topic_alarm_msg.ros_fields__data & topic_alarm_msg.valid)];
+Result: 0
+```
+
+This means that the property does not hold.
+This means our alarm also does not work as expected.
+
+### Task 1.3: Make sure that the battery can't drain below 0
+
+To fix the model, you need to change the `battery_drainer.scxml` file.
+
+In lines 22 to 24, you can see:
+
+```xml
+<ros_rate_callback name="my_timer" target="use_battery">
+    <assign location="battery_percent" expr="battery_percent - 1" />
+</ros_rate_callback>
+```
+
+This gets is a transitions that gets triggered every time the timer `my_timer` fires.
+Because it's target is `use_battery`, the state will not change.
+But as you can see, the battery level is decreased by 1 every time the timer fires.
+We need to make sure that the transition is not taken when the battery level is already 0.
+
+To do this, we can add a condition to the transition:
+Add the attribute `cond="battery_percent > 0"` to the `<ros_rate_callback>` tag.
+
+Now run `as2fm_scxml_to_jani main.xml` and `smc_storm --model main.jani --properties-names battery_over_depleted` again.
```

Comment:
Should these 2 lines be a bulleted list?

Reply:
They could be. But I like it better this way

Reply:
nvm, I changed it. You were right. It also looks visually a little more interesting this way